core/src/test/java/org/elasticsearch/index/query/MissingQueryBuilderTest.java

```diff
@@ -0,0 +1,189 @@
+                    } else {
+                        f = field;
+                    }
+                    filter = fieldNamesFieldType.termQuery(f, context);
+                }
+                // if _field_names are not indexed, we need to go the slow way
+                if (filter == null && fieldType != null) {
+                    filter = fieldType.rangeQuery(null, null, true, true, context);
+                }
+                if (filter == null) {
+                    filter = new TermRangeQuery(field, null, null, true, true);
+                }
+                boolFilter.add(filter, BooleanClause.Occur.SHOULD);
+            }
+
+            existenceFilter = boolFilter;
+            existenceFilter = Queries.not(existenceFilter);;
+        }
+
+        if (nullValue) {
+            for (String field : fields) {
+                MappedFieldType fieldType = context.fieldMapper(field);
+                if (fieldType != null) {
+                    nullFilter = fieldType.nullValueQuery();
+                }
+            }
+        }
+
+        Query filter;
+        if (nullFilter != null) {
+            if (existenceFilter != null) {
+                BooleanQuery combined = new BooleanQuery();
+                combined.add(existenceFilter, BooleanClause.Occur.SHOULD);
+                combined.add(nullFilter, BooleanClause.Occur.SHOULD);
+                // cache the not filter as well, so it will be faster
+                filter = combined;
+            } else {
+                filter = nullFilter;
+            }
+        } else {
+            filter = existenceFilter;
+        }
+
+        if (filter == null) {
+            return null;
+        }
+
+        return new ConstantScoreQuery(filter);
+    }
+
+    @Test
+    public void testValidate() {
+        MissingQueryBuilder missingQueryBuilder = new MissingQueryBuilder("");
+        assertThat(missingQueryBuilder.validate().validationErrors().size(), is(1));
+
+        missingQueryBuilder = new MissingQueryBuilder(null);
+        assertThat(missingQueryBuilder.validate().validationErrors().size(), is(1));
+
+        missingQueryBuilder = new MissingQueryBuilder("field").existence(false).nullValue(false);
+        assertThat(missingQueryBuilder.validate().validationErrors().size(), is(1));
+
+        missingQueryBuilder = new MissingQueryBuilder("field");
+        assertNull(missingQueryBuilder.validate());
+    }
+    
+    @Test
+    public void testBothNullValueAndExistenceFalse() throws IOException {
+        QueryParseContext context = createContext();
+        context.setAllowUnmappedFields(true);
+        try {
+            MissingQueryBuilder.newFilter(context, "field", false, false);
+            fail("MissingQueryBuilder should have failed, both nullValue and existence are false!");
+        } catch (QueryParsingException e) {
+            assertEquals(e.getMessage(), "missing must have either existence, or null_value, or both set to true");
+        }
```

Comment:
Instead of wrapping tested methods that throw exceptions in a try/catch I prefer adding the "expected=exception type" annotation to the test, removes some code and checking the exact error message feels too error prone to me.